pkg/proxy/mapping.go

```diff
@@ -102,8 +174,34 @@ func NewHandler(ctx context.Context, o *proxyoptions.Options, index index.Index)
 
 		handler = WithProxyAuthHeaders(handler, userHeader, groupHeader, extraHeaderPrefix)
 
-		mux.Handle(m.Path, handler)
+		logger.V(2).WithValues("path", m.Path).Info("adding handler")
+		if m.Path == "/" {
+			handlers.defaultHandler = handler
+		} else {
+
+			handlers.mapping = append(handlers.mapping, httpHandlerMapping{
+				weight:  len(m.Path),
+				path:    m.Path,
+				handler: handler,
+			})
+		}
 	}
 
-	return mux, nil
+	handlers.mapping = sortMappings(handlers.mapping)
+
+	return &handlers, nil
+}
+
+func sortMappings(mappings []httpHandlerMapping) []httpHandlerMapping {
+	// sort mappings by weight
+	// higher weight means that the mapping is more specific and should be matched first
+	// Example: /clusters/cluster1/ will be matched before /clusters/
+	for i := range mappings {
+		for j := range mappings {
+			if mappings[i].weight > mappings[j].weight {
+				mappings[i], mappings[j] = mappings[j], mappings[i]
```

Comment:
bubble sort 🎉

Reply:
One and only! Why invest something new when this just does the job :D